2a_model.R

```diff
@@ -119,25 +129,34 @@ p2a_targets_list <- list(
   tar_target(
     p2a_metrics_files,
     {
-    # we need these to make the prepped data file
+    #we need these to make the prepped data file
     p2a_well_obs_data
     
+    #add in the medium observed data
+    p2a_med_obs_data
+    
     base_dir <- "2a_model/src/models"
     snakefile_path <- file.path(base_dir, p2a_model_ids$snakefile_dir, "Snakefile")
     config_path <- file.path(base_dir, p2a_model_ids$config_path)
     # this path is relative to the Snakefile
-    prepped_data_file <- file.path("../../../out/models", p2a_model_ids$model_id, "prepped.npz")
+    prepped_data_file <- file.path("../../../out/models",p2a_model_ids$model_id, "prepped.npz")
+    weights_trained_file <- file.path("../../../out/models",p2a_model_ids$model_id,"nstates_10/nep_100/rep_0/train_weights")
+    output_feather_file <- file.path("../../../out/models",p2a_model_ids$model_id,"nstates_10/nep_100/rep_0/preds.feather")
     
     # First create the prepped data files if they are not already.
     # These are needed to make the predictions.
-    system(stringr::str_glue("snakemake {prepped_data_file} -s {snakefile_path} --configfile {config_path} -j"))
+    system(stringr::str_glue("snakemake {prepped_data_file} -s {snakefile_path} --configfile {config_path} -j -f"))
 
+    system(stringr::str_glue("snakemake {weights_trained_file} -s {snakefile_path} --configfile {config_path} -j -f"))
+    
+    system(stringr::str_glue("snakemake {output_feather_file} -s {snakefile_path} --configfile {config_path} -j -f"))
+    
     # Then touch all of the existing files. This makes the weights "up-to-date"
     # so snakemake doesn't train the models again
-    system(stringr::str_glue("snakemake -s {snakefile_path} --configfile {config_path} -j --touch"))
+    #system(stringr::str_glue("snakemake -s {snakefile_path} --configfile {config_path} -j --touch"))
 
     # then run the snakemake pipeline to produce the predictions and metric files
-    system(stringr::str_glue("snakemake -s {snakefile_path} --configfile {config_path} -j --rerun-incomplete"))
+    system(stringr::str_glue("snakemake -s {snakefile_path} --configfile {config_path} -j -f"))
```

Comment:
If I'm understanding this correctly, one problem with this approach is that it will rerun the full Snakemake pipeline the `p2a_metrics_files` target needs to be run. For example, if I clone this onto my machine, it won't have the metrics file for exp `0`, so targets will trigger the rebuilding of that branch of the `p2a_metrics_files`. Then it will 
 - build the prepped data file, which good (line 142)
 - retrain the `rep_0` model (line 143)
 - make the `rep_0` preds (line 144)
 - build the overall metrics files (line 159) which would include making all of the predictions (unless they are already there, like `rep_0` (line 144))

Reply:
I think the reason you added these lines in here was to force a retraining of the model because the previous trained model weights were different than the new, NHD ones. Is that right? I think a cleaner way to do that would be to just delete the previous model weights and run this target as it was written before. That should trigger a retraining of all of the model replicates etc. Did you try that?

Reply:
Oh yeah, good point! I think this makes sense, I am going through and testing it to make sure it works. One odd behavior that I am running into: When I got through and delete the weights `out/models/0_baseline_LSTM/nstates_10/nep_100/rep_0/train_weights`, then rerun `p2a_metrics_files` using the original set of snakemake commands (`prepped.npz`,`touch`, and `rerun incomplete`), the model produces a new `train_weights` file for all model ids except 0_baseline_LSTM. Even if I delete all the model output, everything within `out/models/0_baseline_LSTM/nstates_10`, I still don't get the model to retrain